File: UVa Online Judge/530.py
def solve():
    n, k = [int(_) for _ in input().split()]

    if not n and not k:
        return 0

    res = 1
    for i in range(max(k, n - k) + 1, n + 1):
        res *= i
    for i in range(1, min(k, n - k) + 1):
        res //= i

    print(res)
    return 1

while (solve()):
    pass

File: LeetCode/530.cpp
class Solution {
    struct Data {
        int minDiff, min, max;
    };
    
    Data rec(TreeNode *n) {
        Data res = { (int)1e9, n->val, n->val };
        if (n->left) {
            Data l = rec(n->left);
            res.min = l.min;
            res.minDiff = min(res.minDiff, min(l.minDiff, n->val - l.max));
        }
        if (n->right) {
            Data r = rec(n->right);
            res.max = r.max;
            res.minDiff = min(res.minDiff, min(r.minDiff, r.min - n->val));
        }
        return res;
    }
    
public:
    int getMinimumDifference(TreeNode *root) {
        return rec(root).minDiff;
    }
};